core/src/percent.rs

```diff
@@ -0,0 +1,52 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021-2023 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Basic struct for working with integer percentages.
+
+use core::cmp::Ord;
+use num_traits::{cast::NumCast, Num};
+use parity_scale_codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+/// Basic struct for working with integer percentages.
```

Comment:
Consider: You don't need this type in core. It's enough to keep just u32 here but introduce `Percent` in `gcore/gstd` in future